lib/apiservers/engine/backends/filter/image.go

```diff
@@ -101,14 +101,18 @@ func IncludeImage(imgFilters filters.Args, listContext *ImageListContext) Filter
 		refs := cache.RepositoryCache().References(listContext.ID)
 		// reference filters
 		refFilters := imgFilters.Get("reference")
+
+		// current filter matches
+		listContext.Tags = make([]string, 0)
```

Comment:
not sure if it makes sense to even bother, but if there is a chance that these arrays will hold at least 1 element, it is better to add some capacity into these array. Let say for 4 elements. However, if there are no elements expected, its better to just keep nil, since nil is treated as a perfect empty array.